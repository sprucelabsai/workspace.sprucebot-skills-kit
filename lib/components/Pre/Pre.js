'use strict';

Object.defineProperty(exports, '__esModule', {
	value: true
});

var _templateObject = _taggedTemplateLiteral(['\n\tfont-family: Courier, \'New Courier\', monospace;\n\tfont-size: 0.8em;\n\ttext-align: left;\n\twhite-space: pre-wrap;\n\twidth: 100%;\n\tword-wrap: break-word;\n\tbackground-color: black;\n\tcolor: green;\n\tpadding: 20px;\n\ttabsize: 1;\n'], ['\n\tfont-family: Courier, \'New Courier\', monospace;\n\tfont-size: 0.8em;\n\ttext-align: left;\n\twhite-space: pre-wrap;\n\twidth: 100%;\n\tword-wrap: break-word;\n\tbackground-color: black;\n\tcolor: green;\n\tpadding: 20px;\n\ttabsize: 1;\n']);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { 'default': obj }; }

function _taggedTemplateLiteral(strings, raw) { return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var _styledComponents = require('styled-components');

var _styledComponents2 = _interopRequireDefault(_styledComponents);

var Pre = _styledComponents2['default'].pre(_templateObject);

exports['default'] = Pre;
module.exports = exports['default'];