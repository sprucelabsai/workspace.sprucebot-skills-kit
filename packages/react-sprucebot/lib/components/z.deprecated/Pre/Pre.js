"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var Pre = function Pre(_ref) {
  var className = _ref.className,
      props = (0, _objectWithoutProperties2.default)(_ref, ["className"]);
  return _react.default.createElement("pre", (0, _extends2.default)({}, props, {
    className: "pre ".concat(className || '')
  }));
};

var _default = Pre;
exports.default = _default;