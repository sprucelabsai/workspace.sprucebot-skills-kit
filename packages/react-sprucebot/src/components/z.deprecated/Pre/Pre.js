import React from 'react'

const Pre = ({ className, ...props }) => {
	return <pre {...props} className={`pre ${className || ''}`} />
}

export default Pre
