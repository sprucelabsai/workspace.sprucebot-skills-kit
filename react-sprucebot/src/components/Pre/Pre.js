import styled from 'styled-components'

const Pre = styled.pre`
	font-family: Courier, 'New Courier', monospace;
	font-size: 0.8em;
	text-align: left;
	white-space: pre-wrap;
	width: 100%;
	word-wrap: break-word;
	background-color: black;
	color: green;
	padding: 20px;
	tabsize: 1;
`

export default Pre
