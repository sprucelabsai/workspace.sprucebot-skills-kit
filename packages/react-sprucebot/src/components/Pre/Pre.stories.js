import React from 'react'
import { storiesOf } from '@storybook/react'
import { withInfo } from '@storybook/addon-info'
import { withKnobs } from '@storybook/addon-knobs/react'
import { withReadme } from 'storybook-readme'
import Pre from './Pre'
import readme from './Pre.md'

const stories = storiesOf('Pre', module)
stories.addDecorator(withKnobs)

stories.add('Interactive', withReadme(readme, () => <Pre />))
