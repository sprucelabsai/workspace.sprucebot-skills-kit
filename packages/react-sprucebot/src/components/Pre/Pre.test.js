import React from 'react'
import Pre from './Pre'
import { shallow } from 'enzyme'

describe('Pre tests', () => {
	it('Should match the snapshot', () => {
		const renderedComponent = shallow(<Pre />)

		expect(renderedComponent).toMatchSnapshot()
	})
})
