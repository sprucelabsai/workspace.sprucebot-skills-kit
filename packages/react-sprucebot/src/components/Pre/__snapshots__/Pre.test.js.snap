// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders 1`] = `
.c0 {
  font-family: Courier,'New Courier',monospace;
  font-size: 0.8em;
  text-align: left;
  white-space: pre-wrap;
  width: 100%;
  word-wrap: break-word;
  background-color: black;
  color: green;
  padding: 20px;
  tabsize: 1;
}

<pre
  className="c0"
/>
`;
